IBAnimatable/GradientDesignable.swift

```diff
@@ -78,14 +46,48 @@ fileprivate extension GradientDesignable where Self: UIView {
   }
 
   func configurePredefinedGradient() -> GradientColor? {
-
     guard let gradientType = predefinedGradient else {
       return nil
     }
     return gradientType.colors
   }
 }
 
+public extension GradientDesignable where Self: UINavigationBar {
+  public func configureGradient() {
+    let predefinedGradient = configurePredefinedGradient()
+    if let startColor = startColor, let endColor = endColor {
+      configureGradient(startColor: startColor, endColor: endColor)
+    } else if let startColor = predefinedGradient?.start, let endColor = predefinedGradient?.end {
+      configureGradient(startColor: startColor, endColor: endColor)
+    }
+  }
+}
+
+fileprivate extension GradientDesignable where Self: UINavigationBar {
```

Comment:
I got it, we try to group extension for `where Self: UIView` or `where Self: UINavigationBar`. I agree with @tbaranes now. That can save to `private extension`s. I hope Swift can support something like 
```
extension GradientDesignable where Self: UIView {
  func foo() {
  }

  private extension {
    func privateFoo() {
    }

  }
}
```

But they don't, we have to use 

```
extension GradientDesignable where Self: UIView {
  func foo() {
  }

  private func privateFoo() {
  }
}
```

@phimage How do you think we use @tbaranes 's practice to group the extensions?